hagglehaul.Server/Controllers/DriverController.cs

```diff
@@ -661,5 +661,36 @@ public async Task<IActionResult> GetAllAvailableTrips([FromBody] TripMarketOptio
             var searchedTrips = await TripsToSearchedTrips(trips);
             return Ok(searchedTrips);
         }
+
+        [Authorize]
+        [HttpPost]
+        [Route("rating")]
+        public async Task<IActionResult> RateRider([FromBody] GiveRating giveRating)
+        {
+            ClaimsPrincipal currentUser = this.User;
+
+            if (currentUser == null) { return BadRequest(new { Error = "Invalid User/Auth" }); };
+
+            var role = currentUser.FindFirstValue(ClaimTypes.Role);
+
+            if (role != "driver") { return Unauthorized(); }
+
+            RiderProfile rider = await _riderProfileService.GetAsync(giveRating.TargetUserEmail);
+            if (rider == null) { return BadRequest(new { Error = "Rider does not exist" }); }
+
+            if (rider.Rating == null)
+                rider.Rating = 0;
+            if (rider.NumRatings == null)
+                rider.NumRatings = 0;
+
+            var totalRatings = rider.Rating * (double)rider.NumRatings;
+
+            rider.NumRatings++;
+
+            rider.Rating = (totalRatings + (double)giveRating.RatingGiven) / (double)rider.NumRatings;
+
+            await _riderProfileService.UpdateAsync(rider.Email, rider);
```

Comment:
per the above comment, update the Trip object here as well (of course, checking that the trip time is 1. in the past 2. has a driver 3. the driver is the driver of the trip)